### PR TITLE
feat: Improve Payment Status Indicators on Invoice Dashboard

### LINKED ISSUE #174: Improve the payment status indicators on the Invoice Dashboard
# Problem

We want to improve our progress indicators when paying a request. Currently the payment progress indicators appear in the bottom left of the invoice details modal, they look like clickable buttons even though they're not, and we can see two on-screen at the same time. 
Image

# Proposed Solution

[Design](https://www.figma.com/design/cEkRPhER2ArCnOwNVDmVVA/Request-Network?node-id=430-2869&t=qlQ28TfwDvEEB7nF-4)
Image

# Source

https://miro.com/app/board/uXjVK_t9NEA=/?moveToWidget=3458764603763997147&cot=14

### PR DESCRIPTION
**Pull request by sstefdev**
Fixes #174 

### Problem
The current payment progress indicators in the Invoice Dashboard have the following issues:
- Appear like clickable buttons despite being non-interactive.
- Multiple indicators can appear on-screen simultaneously, leading to confusion.
- Placement in the bottom left of the invoice details modal affects usability.

### Changes
Indicator Redesign:
- Updated the design to clearly show progress without resembling buttons.
- Improved styling for a modern and accessible appearance.

Behavioral Adjustments:
- Ensured only one progress indicator is visible at a time.
- Added logic to manage and update the indicator's visibility dynamically.

Placement Optimization:
- Moved the indicators to a more intuitive location in the invoice details modal for better visibility.

- Added fix for multiple requests error
- Added ERC-20 approval flow
- Removing Pay Now button when a user signs the transactions

#### How it looks 
https://github.com/user-attachments/assets/eabc7bd1-b327-4c95-bac6-55c65aa181c0

#### Updated (With ERC-20 Approval)
https://github.com/user-attachments/assets/c36a30fd-7b50-4322-8850-3f433a771248

## Summary by CodeRabbit

- **New Features**
	- Enhanced status tracking for payment processes with structured updates.
	- Improved user interface for displaying payment progress with a new list format and icons.

- **Bug Fixes**
	- Updated error handling for unsupported payment networks.

- **Style**
	- Adjusted styling for status lists and items to improve visual representation.

- **Chores**
	- Refactored code for better readability and maintainability.
	- Reformatted variable declarations and reactive statements for clarity.

### CODE REVIEW

**Comment by MantisClone** [comment]
The statuses are in the wrong order. It should check for the correct network first, then get the transaction signature, and finish with the payment detected.

**Comment by MantisClone** [comment]
Previously the first button we showed was switch chain. I think we should change this to Pay Now. Then, we show the payment status stepper and prompt to switch chains if necessary. We also need an intermediate step to check for approval in the case of ERC-20s

File: packages/invoice-dashboard/src/lib/dashboard/invoice-view.svelte
Comment (MantisClone) [inline packages/invoice-dashboard/src/lib/dashboard/invoice-view.svelte:314]: I'm concerned about adding a state boolean dynamically. I agree with CodeRabbit that this should be defined statically.

File: packages/invoice-dashboard/src/lib/dashboard/invoice-view.svelte
Comment (MantisClone) [inline packages/invoice-dashboard/src/lib/dashboard/invoice-view.svelte:340]: I don't like that we must search the `statuses` array each time we want to look up a state boolean.

**Review by MantisClone — COMMENTED** [review]
The state of this component is getting too complex and I think we must refactor it to a state machine that is easier to reason about and maintain. I suspect that error states are not adequately expressed.

That said, I don't want to delay the delivery of this feature so if you can update the video clip on the PR description showing that it works, I'll approve it.

File: packages/invoice-dashboard/src/lib/dashboard/invoice-view.svelte
Comment (sstefdev) [reply packages/invoice-dashboard/src/lib/dashboard/invoice-view.svelte]: Well the problem is that we will see it every time even if we don't have something to approve, so that is why it is added dynamically

File: packages/invoice-dashboard/src/lib/dashboard/invoice-view.svelte
Comment (sstefdev) [reply packages/invoice-dashboard/src/lib/dashboard/invoice-view.svelte]: It is needed, I updated the logic

File: packages/invoice-dashboard/src/lib/dashboard/invoice-view.svelte
Comment (sstefdev) [reply packages/invoice-dashboard/src/lib/dashboard/invoice-view.svelte:418]: Me neither, but that is the best solution for now

File: packages/invoice-dashboard/src/lib/dashboard/invoice-view.svelte
Comment (sstefdev) [reply packages/invoice-dashboard/src/lib/dashboard/invoice-view.svelte:500]: No need, the timeout for testing is removed

**Comment by MantisClone** [comment]
I'm approving to unblock your other work, but I strongly think we must refactor this logic at a later time.

**Review by MantisClone — APPROVED** [review]

File: packages/invoice-dashboard/src/lib/dashboard/invoice-view.svelte
Comment (sstefdev) [reply packages/invoice-dashboard/src/lib/dashboard/invoice-view.svelte:414]: We should be using the index, if we are using the id we will get to the same approach, which is not bad for this small array as I mentioned:
- The find function gives us O(n), which is O(4), very small
- And approach with index is O(1), which is faster, but not noticeable

Comment (sstefdev) [reply packages/invoice-dashboard/src/lib/dashboard/invoice-view.svelte:414]: But will add with index search

File: packages/invoice-dashboard/src/lib/dashboard/invoice-view.svelte
Comment (sstefdev) [reply packages/invoice-dashboard/src/lib/dashboard/invoice-view.svelte:792]: No need for this

**Review by rodrigopavezi — APPROVED** [review]

**Pull request merged**
